iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -596,6 +596,101 @@ static LogicalResult setRootConfig(
       entryPointFn, fftOp, tileSizes, DispatchLoweringPassPipeline::CPUDefault);
 }
 
+static LogicalResult setX86DefaultParallelReductionTileSizes(
+    linalg::GenericOp genericOp, unsigned numLoops,
+    ArrayRef<int64_t> flowTileSizes, ArrayRef<int64_t> minTileSizes,
+    ArrayRef<int64_t> maxTileSizes, SmallVectorImpl<int64_t> &parallelTileSizes,
+    SmallVectorImpl<int64_t> &reductionTileSizes) {
+  if (getLoweringConfig(genericOp)) return success();
+
+  parallelTileSizes.append(numLoops, 0);
+  Optional<SmallVector<int64_t, 4>> staticLoopRanges =
+      genericOp.getStaticLoopRanges();
+  for (auto loopNum : llvm::seq<unsigned>(0, numLoops)) {
+    if (flowTileSizes[loopNum]) {
+      parallelTileSizes[loopNum] =
+          getMaxTileSize(0, flowTileSizes[loopNum], minTileSizes[loopNum],
+                         minTileSizes[loopNum]);
+    } else {
+      // If the flow level tile size is zero, and static loop range is 0 as
+      // well, set the tile sizes here to zero as well.
+      parallelTileSizes[loopNum] =
+          (staticLoopRanges && staticLoopRanges.getValue()[loopNum] == 1)
+              ? 0
+              : minTileSizes[loopNum];
+    }
+  }
+
+  splitParallelAndReductionTiles(genericOp, parallelTileSizes,
+                                 reductionTileSizes);
+  return success();
+}
+
+/// Returns true if the operation is a GenericOp implementing a supported
+/// transposition.
+static bool isSupportedTransposeOp(linalg::GenericOp genericOp) {
+  // Check that the op has at least 2 dimensions.
+  if (genericOp.getNumLoops() < 2) return false;
+
+  // Check that the op has only one input and one output.
+  // TODO(diegocaballero): Generalize to multiple inputs.
+  if ((genericOp.getNumInputs() != 1) || (genericOp.getNumOutputs() != 1))
+    return false;
```

Comment:
style nit: remove redundant braces.  In IREE coding style, we wrap if/or statements with braces even it is a single line.

```suggestion
  if (genericOp.getNumInputs() != 1 || genericOp.getNumOutputs() != 1) {
    return false;
  }
```